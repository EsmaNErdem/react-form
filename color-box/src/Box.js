
const Box = ({id, color, width, height, handleRemove}) => {
    return (
        <>
        <div 
        className="Box" 
        style={{background: color, width: width, height: height}}>
        </div>
        <button data-testid={color} onClick={() => handleRemove(id)}>X</button>
        </>
    )
}

export default Box;