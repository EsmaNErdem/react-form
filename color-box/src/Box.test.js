import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Box from "./Box";

test("renders without crashing", () => {
    render(<Box />);
});

test("it renders and matches with snaphot", () => {
    const { asFragment } = render(<Box />);
    expect(asFragment()).toMatchSnapshot();
});