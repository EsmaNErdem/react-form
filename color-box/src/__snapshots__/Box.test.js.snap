// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches with snaphot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  />
  <button>
    X
  </button>
</DocumentFragment>
`;
